lib/cog/relay/info.ex

```diff
@@ -46,24 +55,22 @@ defmodule Cog.Relay.Info do
   # Private functions
 
   defp info(%{"list_bundles" => %{"relay_id" => relay_id, "reply_to" => reply_to}}, state) do
-    case Repo.one(Queries.Relay.for_id(relay_id)) do
-      %Relay{}=relay ->
-        Enum.flat_map(relay.groups, &(&1.bundles))
-        |> prepare_bundles
-      nil ->
-        []
+    all = fn(:get, data, next) ->
+      Enum.flat_map(data, &next.(Map.delete(&1, :__struct__)))
     end
-    |> respond(reply_to, state)
-  end
 
-  defp prepare_bundles(bundles) do
-    prepared_bundles = Enum.map(bundles, fn(bundle) ->
-      %{name: bundle.name,
-        config_file: bundle.config_file,
-        enabled: bundle.enabled}
-    end)
+    case Repo.get(Relay, relay_id) do
+      %Relay{}=relay ->
+        relay = Repo.preload(relay, [groups: :bundles])
+
+        bundles = get_in(relay.groups, [all, :bundles])
+        |> Enum.map(&Map.take(&1, [:name, :config_file, :enabled]))
 
-    %{bundles: prepared_bundles}
+        respond(%{bundles: bundles}, reply_to, state)
+      nil ->
+        ## If we get a nil back then the relay isn't registered with Cog.
+        respond(%{error: "Relay with id #{relay_id} was not recognized."}, reply_to, state)
```

Comment:
I don't @christophermaier since we never actually hit this point in the code with a real relay. I suppose I could dink with bits to force it to happen, but I'm not sure that's worth doing here.